tools/ramdump/ramdump_tool.c

```diff
@@ -328,11 +330,48 @@ static int ramdump_recv(int dev_fd)
 				}
 			}
 			printf("]\n");
+
+			ret = read(dev_fd, &buf, 1);
+			if (ret != 1) {
+				printf("Receiving ramdump %dTH byte failed, ret = %d\n", count, ret);
+				fclose(bin_fp);
+				return -1;
+			}
+			ret = fwrite(&buf, 1, 1, bin_fp);
```

Comment:
As per my experience, streaming is always slower than POSIX, you have any reason why use fwrite instead of write?
you are already using 'read' in line 334 with another file descriptor...